lib/ssf/client.rb

```diff
@@ -39,21 +39,27 @@ def send_to_socket(span)
     end
 
     def start_span(name: '', tags: {}, parent: nil, indicator: false)
+      tags = Ssf::SSFSpan.clean_tags(tags)
       if parent
-        new_tags = {}
-        parent.tags.each do |key, value|
-          if key != 'name'
-            new_tags[key] = value
-          end
-        end
-        new_tags = Ssf::SSFSpan.clean_tags(tags.merge(new_tags))
-        start_span_from_context(name: name, tags: new_tags, trace_id: parent.trace_id, parent_id: parent.id, indicator: indicator)
+        start_span_from_context(
+          name: name,
+          tags: Hash[parent.tags].merge!(tags),
+          trace_id: parent.trace_id,
+          parent_id: parent.id,
+          indicator: indicator,
+          clean_tags: false,
```

Comment:
Right now, `clean_tags` is never going to be true. Is that what we want?

Reply:
Yes. Any code that calls `start_span_from_context` directly will have the tags 'cleaned', but specifying the option allows us to avoid re-cleaning them multiple times in other cases. Weirdly, protobuf won't accept a protobuf Map for a map field, so we have to turn it back into a hash in some cases too.

Reply:
(In both cases where `start_span` calls `start_span_from_context`, the tags are pre-cleaned, which is why they both send false)

Reply:
Ah, I see. Is performance the only concern with not cleaning twice?

Reply:
I expect there should be some small performance gain at least, though I'm not sure how much of a difference there will be. I generally don't like to write code that does redundant-looking things because it winds up looking like a mistake that people reading the code will question. If it's clunky like that, it's a sign that it should be phrased better. I settled for a compromise -- making it explicit/intentional what's going on rather than trying to fully fix it -- since the PR's focus isn't on performance/cleanup, but simply fixing the problem.

Reply:
Ah, I agree; I just wanted to check if that was the only reason, or if there was a correctness issue involved.